conda/recipes/rapids/meta.yaml

```diff
@@ -54,6 +54,12 @@ requirements:
     - ptxcompiler  # CUDA enhanced compat. See https://github.com/rapidsai/ptxcompiler
     {% endif %}
     - conda-forge::ucx {{ ucx_version }}
+    # TODO: remove pins on pyogrio and tiledb once cuspatial supports 'fmt>11' and 'spdlog>14'
+    # ref:
+    #   * https://github.com/rapidsai/build-planning/issues/56
+    #   * https://github.com/rapidsai/cuspatial/pull/1453#issuecomment-2335527542
+    - pyogrio <0.8
+    - tiledb <2.19
```

Comment:
It looks to me like it's taking FOREVER (2+ hours) for conda to solve the runtime environment for the `rapids` package on Python 3.12.

I strongly suspect that the root cause is "RAPIDS is pinned to an old range of `fmt` / `spdlog` versions that causes the solver to have to backtrack a lot to find older versions of spatial dependencies".

The same issue we saw in https://github.com/rapidsai/cuspatial/pull/1453#issuecomment-2335527542.

I found that when I explicitly pinned `pyogrio` and `tiledb` to versions that were successfully found in recent `cuspatial` conda test jobs ([build link](https://github.com/rapidsai/cuspatial/actions/runs/10798574225/job/29953180637#step:7:491)), I can get a successful solve in just a few minutes.

I think doing that here should be ok... this is just helping the solver get to a workable solution faster, and it could be removed once we update the `fmt` / `spdlog` pins across RAPIDS (which is in-progress in https://github.com/rapidsai/build-planning/issues/56 and will hopefully be done within the 24.10 release).

Tried the following, pinning a bunch of RAPIDS libraries to their latest nightly versions (to avoid backtracking to other pre-Python-3.12 nightlies), on an x86_64 machine with CUDA 12.2.

```shell
conda create \
    --name delete-me \
    --dry-run \
    -v \
    --override-channels \
    --channel rapidsai \
    --channel rapidsai-nightly \
    --channel conda-forge \
    --channel nvidia \
        'cuda-version=12.2' \
        'cudf==24.10.00a320' \
        'cugraph==24.10.00a74' \
        'cuml==24.10.00a55' \
        'cuproj==24.10.00a41' \
        'cupy==13.3.0' \
        'cuspatial==24.10.00a41' \
        'cuml==24.10.00a60' \
        'cuxfilter==24.10.00a19' \
        'dask-cudf==24.10.00a320' \
        'libcudf==24.10.00a320' \
        'libraft==24.10.00a32' \
        'libraft-headers==24.10.00a32' \
        'librmm==24.10.00a38' \
        'nccl==2.22.3.1' \
        'numba==0.60.0' \
        'numpy==1.26.4' \
        'nvtx==0.2.10' \
        'pylibcudf==24.10.00a320' \
        'pylibraft==24.10.00a32' \
        'pynvjitlink==0.3.0' \
        'pyogrio<0.8' \
        'python=3.12' \
        'rmm==24.10.00a38' \
        'tiledb<2.19' \
        'conda-forge::ucx==1.17.0' \
        'ucx-py==0.40.00a12'
```

Reply:
With just the `tildeb` + `pyogrio` pins alone here, I still saw 3 of the 4 Python 3.12 builds not making progress after 45 minutes. Just pushed https://github.com/rapidsai/integration/pull/719/commits/57edadadcbbdd027d1f5476f20182e5871d1cda9 putting floors on all the RAPIDS libraries, to see if preventing the solver from falling back to earlier nightlies helps.

It should cut the search space significantly, I hope... there have been 320 `cudf` releases within the 24.10 cycle so far, for example.

Reply:
Looks like the combination of those two changes helped!

`conda` was able to solve the runtime environment for all the environments we test here: https://github.com/rapidsai/integration/actions/runs/10891738111/job/30223298780?pr=719.

In a manageable amount of time:

* `(amd64, 3.12, 11.8.0, ubuntu22.04)` = 30m33s
* `(amd64, 3.12, 12.5.1, ubuntu22.04)` = 17m36s
* `(arm64, 3.12, 11.8.0, ubuntu22.04)` = 23m08s
* `(arm64, 3.12, 12.5.1, ubuntu22.04)` = 19m39s